### PR TITLE
fix: load providers from //:providers.bzl

### PR DESCRIPTION
**Pull request by manekinekko**
**BREAKING CHANGE**: `@build_bazel_rules_nodejs//:declaration_provider.bzl` has been deleted; load from `@build_bazel_rules_nodejs//:providers.bzl` instead. See https://github.com/bazelbuild/rules_nodejs/commit/cc648183160832b1540c3893963e8ef4a45e4220

### CODE REVIEW

**Comment by UebelAndre** [comment]
Thanks for fixing this! I think you'll also need to update the `docs/BUILD.bazel`:
https://github.com/bazelbuild/rules_rust/blob/e589105b4e8181dd1d0d8ccaa0cf3267efb06e86/docs/BUILD.bazel#L7-L19

File: docs/BUILD.bazel [hunk 1]
```diff
@@ -8,7 +8,7 @@ bzl_library(
     name = "docs_deps",
     srcs = [
         "@bazel_tools//tools:bzl_srcs",
-        "@build_bazel_rules_nodejs//internal/providers:bzl",
+        "@build_bazel_rules_nodejs//:providers:bzl",
```
Comment (UebelAndre) [inline docs/BUILD.bazel:11]: Typo maybe?
```suggestion
        "@build_bazel_rules_nodejs//:providers.bzl",
```

**Review by UebelAndre — CHANGES_REQUESTED** [review]

**Comment by UebelAndre** [comment]
It seems the rolling release builds are unable to start? Doesn't seem like this change would impact that.
cc @hlopko @philwo

**Comment by philwo** [comment]
macOS workers do not yet have a new enough version of Bazelisk yet to support rolling releases. Will fix that next week.

RBE Rolling Bazel seems like some issue with the container config maybe. @coeuvre any idea?

File: docs/BUILD.bazel (hunk 1 above)
Comment (manekinekko) [reply docs/BUILD.bazel]: Good catch! Thank you.

**Comment by UebelAndre** [comment]
@manekinekko Seems like `providers.bzl` is not publicly visible?
```
in bzl_library rule //:docs_deps: target '@build_bazel_rules_nodejs//:providers.bzl' is not visible from target '//:docs_deps'. Check the visibility declaration of the former target if you think the dependency is legitimate. To set the visibility of that source file target, use the exports_files() function
```
Maybe that's an oversight in `rules_nodejs`?

**Comment by manekinekko** [comment]
Weird! I am using v4.2.0. Maybe that's why it's working for me.

```bazel
http_archive(
    name = "build_bazel_rules_nodejs",
    sha256 = "4e1a5633267a0ca1d550cced2919dd4148575c0bafd47608b88aea79c41b5ca3",
    urls = ["https://github.com/bazelbuild/rules_nodejs/releases/download/4.2.0/rules_nodejs-4.2.0.tar.gz"],
)
```

cc @alexeagle @gregmagolan

**Comment by alexeagle** [comment]
Hi @manekinekko nice to see you again :)

```
ERROR: /workdir/docs/BUILD.bazel:7:12: in bzl_library rule //:docs_deps: target '@build_bazel_rules_nodejs//:providers.bzl' is not visible from target '//:docs_deps'. Check the visibility declaration of the former target if you think the dependency is legitimate. To set the visibility of that source file target, use the exports_files() function
```
this is because you're referencing the providers.bzl file directly, instead you should use some `filegroup` target in that package that exposes all the files (probably named "bzl")

**Comment by manekinekko** [comment]
Thank you for the hint @alexeagle ❤️

@UebelAndre looks the `bzl_library` rule needs to be updated as well. Do you need to fix that?

https://github.com/bazelbuild/rules_rust/blob/e589105b4e8181dd1d0d8ccaa0cf3267efb06e86/docs/BUILD.bazel#L07-L12

**Comment by coeuvre** [comment]
RBE Rolling Bazel issue is fixed by https://github.com/bazelbuild/rules_rust/pull/958.

**Comment by UebelAndre** [comment]
Closing this in favor of https://github.com/bazelbuild/rules_rust/pull/984

**Pull request closed without merging**